physical/oci_objectstorage/oci_objectstorage.go

```diff
@@ -0,0 +1,398 @@
+		ObjectName:         &entry.Key,
+		ContentLength:      &size,
+		PutObjectBody:      ioutil.NopCloser(bytes.NewReader(entry.Value)),
+		OpcMeta:            nil,
+		OpcClientRequestId: &opcClientRequestId,
+	}
+
+	resp, err := o.client.PutObject(ctx, request)
+	o.logRequest("put", resp.RawResponse, resp.OpcClientRequestId, resp.OpcRequestId, err)
+
+	if err != nil {
+		metrics.SetGauge(metricPutFailed, 1)
+
+		if resp.RawResponse != nil {
+			defer resp.RawResponse.Body.Close()
+		}
+
+		return errwrap.Wrapf("failed to put data: {{err}}", err)
+	}
+	o.logger.Debug("physical/oci: put end")
+
+	defer resp.RawResponse.Body.Close()
+
+	return nil
+}
+
+func (o *Backend) Get(ctx context.Context, key string) (*physical.Entry, error) {
+	o.logger.Debug("physical/oci: get")
+	defer metrics.MeasureSince(metricGetFull, time.Now())
+	metrics.SetGauge(metricPermitsUsed, float32(o.permitPool.CurrentPermits()))
+	startAcquirePool := time.Now()
+	o.permitPool.Acquire()
+	defer o.permitPool.Release()
+	metrics.MeasureSince(metricGetAcquirePool, startAcquirePool)
+
+	defer metrics.MeasureSince(metricGet, time.Now())
+	opcClientRequestId, err := uuid.GenerateUUID()
+	if err != nil {
+		o.logger.Debug("physical/oci: Get: error generating UUID")
+		opcClientRequestId = ""
+	}
+	o.logger.Debug("physical/oci: Get", "opc-client-request-id", opcClientRequestId)
+	request := objectstorage.GetObjectRequest{
+		NamespaceName:      &o.namespaceName,
+		BucketName:         &o.bucketName,
+		ObjectName:         &key,
+		OpcClientRequestId: &opcClientRequestId,
+	}
+
+	response, err := o.client.GetObject(ctx, request)
+	o.logRequest("get", response.RawResponse, response.OpcClientRequestId, response.OpcRequestId, err)
+
+	if err != nil {
+
+		if response.RawResponse != nil && response.RawResponse.StatusCode == http.StatusNotFound {
+			defer response.RawResponse.Body.Close()
+			return nil, nil
+		}
+
+		metrics.SetGauge(metricGetFailed, 1)
+		return nil, errwrap.Wrapf(fmt.Sprintf("failed to read Value for %q: {{err}}", key), err)
+	}
+
+	defer response.RawResponse.Body.Close()
+
+	body, err := ioutil.ReadAll(response.Content)
+	if err != nil {
+		metrics.SetGauge(metricGetFailed, 1)
+		return nil, errwrap.Wrapf("failed to decode Value into bytes: {{err}}", err)
+	}
+
+	o.logger.Debug("physical/oci: get end")
+
+	return &physical.Entry{
+		Key:   key,
+		Value: body,
+	}, nil
+}
+
+func (o *Backend) Delete(ctx context.Context, key string) error {
+	o.logger.Debug("physical/oci: delete")
+	defer metrics.MeasureSince(metricDeleteFull, time.Now())
+	metrics.SetGauge(metricPermitsUsed, float32(o.permitPool.CurrentPermits()))
+	startAcquirePool := time.Now()
+	o.permitPool.Acquire()
+	defer o.permitPool.Release()
+	metrics.MeasureSince(metricDeleteAcquirePool, startAcquirePool)
+
+	defer metrics.MeasureSince(metricDelete, time.Now())
+	opcClientRequestId, err := uuid.GenerateUUID()
+	if err != nil {
+		o.logger.Debug("physical/oci: Delete: error generating UUID")
```

Comment:
I'd just return the `err` here. Not being able to get a random number is indicative of low level problems and we'd want to fail the request.